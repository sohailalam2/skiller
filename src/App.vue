<template>
  <div id="app">
    <router-view/>
  </div>
</template>

<script>
import { mapActions } from 'vuex';

export default {
  // initialize the local store with all skills
  async beforeMount() {
    await this.doInitialize();
  },
  methods: {
    ...mapActions(['doInitialize']),
  },
};
</script>

<style lang="scss">
  @import "./assets/main";
</style>
