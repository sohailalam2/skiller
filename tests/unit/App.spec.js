import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';

import App from '@/App.vue';

const localVue = createLocalVue();
const router = new VueRouter();

localVue.use(VueRouter);
localVue.use(Vuex);

describe('App.vue', () => {
  const storeProps = {
    actions: { doInitialize: jest.fn() },
  };
  let store;
  beforeEach(() => {
    store = new Vuex.Store(storeProps);
  });

  it('render the app', () => {
    const wrapper = shallowMount(App, { localVue, router, store });
    expect(storeProps.actions.doInitialize).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  });
});
