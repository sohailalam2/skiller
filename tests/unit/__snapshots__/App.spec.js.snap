// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue render the app 1`] = `
<div
  id="app"
>
  <router-view-stub
    name="default"
  />
</div>
`;
